File: algorithms/easy/67_add_binary.py
# 67. Add Binary

def add_binary(a, b):
    """Add two binary strings.
    
    Args:
        a: First binary string
        b: Second binary string
    
    Returns:
        String representing the sum of a and b in binary
    """
    # Convert binary strings to integers, add them, then convert back to binary
    # The [2:] slice removes the '0b' prefix from Python's binary representation
    return bin(int(a, 2) + int(b, 2))[2:]

# Alternative implementation without built-in conversion
def add_binary_manual(a, b):
    result = ""
    # Start from the end of both strings
    i, j = len(a) - 1, len(b) - 1
    carry = 0
    
    # Process both strings from right to left
    while i >= 0 or j >= 0 or carry:
        # Get the current bits, or 0 if we've processed all bits
        bit_a = int(a[i]) if i >= 0 else 0
        bit_b = int(b[j]) if j >= 0 else 0
        
        # Calculate the sum and carry
        sum_val = bit_a + bit_b + carry
        result = str(sum_val % 2) + result  # Add the current bit to the result
        carry = sum_val // 2  # Update the carry
        
        # Move to the next bits
        i -= 1
        j -= 1
    
    return result

# Test cases
print(add_binary("11", "1"))       # "100"
print(add_binary("1010", "1011"))  # "10101"

File: algorithms/easy/67_add_binary.js
/**
 * 67. Add Binary
 * 
 * @param {string} a
 * @param {string} b
 * @return {string}
 */
var addBinary = function(a, b) {
    // Using built-in methods (may not work for very large binary strings due to JS number limitations)
    return (BigInt('0b' + a) + BigInt('0b' + b)).toString(2);
};

// Alternative implementation without built-in conversion
var addBinaryManual = function(a, b) {
    let result = "";
    // Start from the end of both strings
    let i = a.length - 1;
    let j = b.length - 1;
    let carry = 0;
    
    // Process both strings from right to left
    while (i >= 0 || j >= 0 || carry > 0) {
        // Get the current bits, or 0 if we've processed all bits
        const bitA = i >= 0 ? parseInt(a[i]) : 0;
        const bitB = j >= 0 ? parseInt(b[j]) : 0;
        
        // Calculate the sum and carry
        const sum = bitA + bitB + carry;
        result = (sum % 2) + result;  // Add the current bit to the result
        carry = Math.floor(sum / 2);  // Update the carry
        
        // Move to the next bits
        i--;
        j--;
    }
    
    return result;
};

// Test cases
console.log(addBinary("11", "1"));       // "100"
console.log(addBinary("1010", "1011"));  // "10101"